compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -7134,25 +7136,77 @@ private void createTypeCastExprForRelationalExpr(BLangBinaryExpr binaryExpr, int
                                                                 int rhsExprTypeTag) {
         boolean isLhsIntegerType = TypeTags.isIntegerTypeTag(lhsExprTypeTag);
         boolean isRhsIntegerType = TypeTags.isIntegerTypeTag(rhsExprTypeTag);
+        BType lhsExprType = binaryExpr.lhsExpr.getBType();
```

Comment:
Don't we already have these information as arguments to this function `rhsExprTypeTag` and `lhsExprTypeTag`?
Please reuse if possible